test/unit/workflows/test_refactor_models.py

```diff
@@ -27,41 +40,52 @@ def test_root_list():
     actions = ar.actions
 
     a0 = actions[0]
-    assert a0.tool_state["a"] == 6
-    assert a0.label == "foobar"
+    assert a0.action_type == "add_step"
+    a0t = cast(AddStepAction, a0)
+    assert a0t.tool_state
+    assert a0t.tool_state["a"] == 6
+    assert a0t.label == "foobar"
```

Comment:
What about asserting the type instead?

```suggestion
    assert isinstance(a0, AddStepAction)
    assert a0.tool_state
    assert a0.tool_state["a"] == 6
    assert a0.label == "foobar"
```

Or even with the cast, you don't need to assign to a different variable name.

Same below.